src/PIL/DdsImagePlugin.py

```diff
@@ -346,34 +346,27 @@ def _open(self):
         # pixel format
         pfsize, pfflags, fourcc, bitcount = struct.unpack("<4I", header.read(16))
         masks = struct.unpack("<4I", header.read(16))
-        if flags & DDSD.CAPS:
-            header.seek(20, io.SEEK_CUR)
         n = 0
         rawmode = None
         if pfflags & DDPF.RGB:
             # Texture contains uncompressed RGB data
             masks = {mask: ["R", "G", "B", "A"][i] for i, mask in enumerate(masks)}
-            if bitcount == 24:
+            if bitcount == 8:
+                self._mode = "L"
+            elif bitcount == 24:
                 self._mode = "RGB"
-                rawmode = masks[0x00FF0000] + masks[0x0000FF00] + masks[0x000000FF]
+                rawmode = masks[0x000000FF] + masks[0x0000FF00] + masks[0x00FF0000]
             elif bitcount == 32 and pfflags & DDPF.ALPHAPIXELS:
                 self._mode = "RGBA"
                 rawmode = (
-                    masks[0xFF000000]
-                    + masks[0x00FF0000]
+                    masks[0x000000FF]
                     + masks[0x0000FF00]
-                    + masks[0x000000FF]
+                    + masks[0x00FF0000]
+                    + masks[0xFF000000]
                 )
             else:
                 msg = f"Unsupported bitcount {bitcount} for {pfflags}"
                 raise OSError(msg)
-            rawmode = rawmode[::-1]
-        elif pfflags & DDPF.ALPHA:
-            if bitcount == 8:
-                self._mode = "L"
-            else:
-                msg = f"Unsupported bitcount {bitcount} for {pfflags}"
-                raise OSError(msg)
```

Comment:
Why alpha block is gone?

Reply:
The ALPHA block only supports a bitcount of 8. With the other changes here, we now support bitcount of 8 for RGB and LUMINANCE.

https://learn.microsoft.com/en-us/windows/win32/direct3ddds/dds-pixelformat states that bitcount is

> Valid when dwFlags includes DDPF_RGB, DDPF_LUMINANCE, or DDPF_YUV.

So only YUV is left - I don't think we should support YUV without actually transforming the color, so ALPHA can be removed.